aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -241,6 +245,33 @@ private List<String> deleteObjectsInBucket(String bucket, Collection<String> obj
     return Lists.newArrayList();
   }
 
+  @Override
+  public Iterable<FileInfo> listPrefix(String prefix) {
+    S3URI s3uri = new S3URI(prefix, awsProperties.s3BucketToAccessPointMapping());
+    ListObjectsV2Request request = ListObjectsV2Request.builder().bucket(s3uri.bucket()).prefix(s3uri.key()).build();
+
+    return () -> client().listObjectsV2Paginator(request).stream()
+        .flatMap(r -> r.contents().stream())
+        .map(o -> new FileInfo(
+            String.format("%s://%s/%s", s3uri.scheme(), s3uri.bucket(), o.key()),
```

Comment:
Minor: The bucket returned by `S3URI` may actually be an access point reference, which could break URI parsing if this were to embed it in a location.

I think the solution is to move the S3 access point mapping out of `S3URI`, since that class should remain simple and report what was parsed and avoid this kind of issue. Let's not fix it here, but we should revisit this.

FYI @jackye1995.